File: src/main/java/baboon/industry/recipe/RecipesMacerator.java
package baboon.industry.recipe;

import baboon.industry.block.I2Blocks;
import baboon.industry.item.I2Items;
import net.minecraft.core.block.Block;
import net.minecraft.core.item.Item;
import net.minecraft.core.item.ItemStack;

import java.util.HashMap;

public class RecipesMacerator {
    private static final HashMap<Integer, ItemStack> recipeList = new HashMap<>();

    public RecipesMacerator() {
        addRecipe(I2Blocks.blockTin.id, new ItemStack(I2Items.dustTin, 9));
        addRecipe(I2Blocks.oreTinStone.id, new ItemStack(I2Items.dustTin, 2));
        addRecipe(I2Blocks.oreTinBasalt.id, new ItemStack(I2Items.dustTin, 2));
        addRecipe(I2Blocks.oreTinLimestone.id, new ItemStack(I2Items.dustTin, 2));
        addRecipe(I2Blocks.oreTinGranite.id, new ItemStack(I2Items.dustTin, 2));
        addRecipe(I2Items.oreRawTin.id, new ItemStack(I2Items.dustTin, 2));
        addRecipe(I2Items.ingotTin.id, new ItemStack(I2Items.dustTin, 1));

        addRecipe(I2Blocks.blockCopper.id, new ItemStack(I2Items.dustCopper, 9));
        addRecipe(I2Blocks.oreCopperStone.id, new ItemStack(I2Items.dustCopper, 2));
        addRecipe(I2Blocks.oreCopperBasalt.id, new ItemStack(I2Items.dustCopper, 2));
        addRecipe(I2Blocks.oreCopperLimestone.id, new ItemStack(I2Items.dustCopper, 2));
        addRecipe(I2Blocks.oreCopperGranite.id, new ItemStack(I2Items.dustCopper, 2));
        addRecipe(I2Items.oreRawCopper.id, new ItemStack(I2Items.dustCopper, 2));
        addRecipe(I2Items.ingotCopper.id, new ItemStack(I2Items.dustCopper, 1));

        addRecipe(I2Blocks.blockBronze.id, new ItemStack(I2Items.dustBronze, 9));
        addRecipe(I2Items.ingotBronze.id, new ItemStack(I2Items.dustBronze, 1));

        addRecipe(Block.blockIron.id, new ItemStack(I2Items.dustIron, 9));
        addRecipe(Block.oreIronStone.id, new ItemStack(I2Items.dustIron, 2));
        addRecipe(Block.oreIronBasalt.id, new ItemStack(I2Items.dustIron, 2));
        addRecipe(Block.oreIronLimestone.id, new ItemStack(I2Items.dustIron, 2));
        addRecipe(Block.oreIronGranite.id, new ItemStack(I2Items.dustIron, 2));
        addRecipe(Item.oreRawIron.id, new ItemStack(I2Items.dustIron, 2));
        addRecipe(Item.ingotIron.id, new ItemStack(I2Items.dustIron, 1));

        addRecipe(Block.blockGold.id, new ItemStack(I2Items.dustGold, 9));
        addRecipe(Block.oreGoldStone.id, new ItemStack(I2Items.dustGold, 2));
        addRecipe(Block.oreGoldBasalt.id, new ItemStack(I2Items.dustGold, 2));
        addRecipe(Block.oreGoldLimestone.id, new ItemStack(I2Items.dustGold, 2));
        addRecipe(Block.oreGoldGranite.id, new ItemStack(I2Items.dustGold, 2));
        addRecipe(Item.oreRawGold.id, new ItemStack(I2Items.dustGold, 2));
        addRecipe(Item.ingotGold.id, new ItemStack(I2Items.dustGold, 1));

        addRecipe(Block.blockCoal.id, new ItemStack(I2Items.dustCoal, 8));
        addRecipe(Item.coal.id, new ItemStack(I2Items.dustCoal, 1));
    }

    public static void addRecipe(int input, ItemStack output) {
        recipeList.put(input, output);
    }

    public ItemStack getRecipeResult(int i) {
        return recipeList.get(i);
    }

    public HashMap<Integer, ItemStack> getRecipeList() {
        return recipeList;
    }
}

File: src/main/kotlin/turniplabs/industry/recipes/RecipesMacerator.kt
package turniplabs.industry.recipes

import net.minecraft.core.block.Block
import net.minecraft.core.item.Item
import net.minecraft.core.item.ItemStack
import turniplabs.industry.blocks.IndustryBlocks
import turniplabs.industry.items.IndustryItems

object RecipesMacerator {
    private val recipeList = HashMap<Any?, Any?>()

    init {
        addRecipe(IndustryBlocks.oreCopperStone, IndustryItems.copperDust, 2)
        addRecipe(IndustryBlocks.oreCopperBasalt, IndustryItems.copperDust, 2)
        addRecipe(IndustryBlocks.oreCopperLimestone, IndustryItems.copperDust, 2)
        addRecipe(IndustryBlocks.oreCopperGranite, IndustryItems.copperDust, 2)
        addRecipe(IndustryBlocks.oreTinStone, IndustryItems.tinDust, 2)
        addRecipe(IndustryBlocks.oreTinBasalt, IndustryItems.tinDust, 2)
        addRecipe(IndustryBlocks.oreTinLimestone, IndustryItems.tinDust, 2)
        addRecipe(IndustryBlocks.oreTinGranite, IndustryItems.tinDust, 2)
        addRecipe(Block.oreIronStone, IndustryItems.ironDust, 2)
        addRecipe(Block.oreIronBasalt, IndustryItems.ironDust, 2)
        addRecipe(Block.oreIronLimestone, IndustryItems.ironDust, 2)
        addRecipe(Block.oreIronGranite, IndustryItems.ironDust, 2)
        addRecipe(Block.oreGoldStone, IndustryItems.goldDust, 2)
        addRecipe(Block.oreGoldBasalt, IndustryItems.goldDust, 2)
        addRecipe(Block.oreGoldLimestone, IndustryItems.goldDust, 2)
        addRecipe(Block.oreGoldGranite, IndustryItems.goldDust, 2)

        addRecipe(IndustryBlocks.copperBlock, IndustryItems.copperDust, 9)
        addRecipe(IndustryBlocks.tinBlock, IndustryItems.tinDust, 9)
        addRecipe(IndustryBlocks.bronzeBlock, IndustryItems.bronzeDust, 9)
        addRecipe(Block.blockIron, IndustryItems.ironDust, 9)
        addRecipe(Block.blockGold, IndustryItems.goldDust, 9)
        addRecipe(Block.blockCoal, IndustryItems.coalDust, 8)

        addRecipe(IndustryItems.rawCopperOre, IndustryItems.copperDust, 2)
        addRecipe(IndustryItems.rawTinOre, IndustryItems.tinDust, 2)
        addRecipe(Item.oreRawIron, IndustryItems.ironDust, 2)
        addRecipe(Item.oreRawGold, IndustryItems.goldDust, 2)
        addRecipe(IndustryItems.copperIngot, IndustryItems.copperDust, 1)
        addRecipe(IndustryItems.tinDust, IndustryItems.tinDust, 1)
        addRecipe(IndustryItems.bronzeIngot, IndustryItems.bronzeDust, 1)
        addRecipe(Item.ingotIron, IndustryItems.ironDust, 1)
        addRecipe(Item.ingotGold, IndustryItems.goldDust, 1)
        addRecipe(Item.coal, IndustryItems.coalDust, 1)
    }

    fun addRecipe(itemInput: Item, outputItem: Item, outputCount: Int) {
        recipeList[itemInput.id] = ItemStack(outputItem, outputCount)
    }

    fun addRecipe(itemInput: Item, outputBlock: Block, outputCount: Int) {
        recipeList[itemInput.id] = ItemStack(outputBlock, outputCount)
    }

    fun addRecipe(blockInput: Block, outputItem: Item, outputCount: Int) {
        recipeList[blockInput.id] = ItemStack(outputItem, outputCount)
    }

    fun addRecipe(blockInput: Block, outputBlock: Block, outputCount: Int) {
        recipeList[blockInput.id] = ItemStack(outputBlock, outputCount)
    }

    fun getResult(i: Int): ItemStack? {
        return recipeList[i] as ItemStack?
    }

    fun getRecipeList(): HashMap<Any?, Any?> {
        return recipeList
    }
}